server/generate.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/sh
+
+set -e
+
+package="LeanCode.ContractsGenerator.${SERVER_VERSION:=1.0.2}.zip"
+source="https://github.com/leancodepl/contractsgenerator/releases/download/v${SERVER_VERSION}/${package}"
+dest="${XDG_STATE_HOME:-${HOME}/.local/state}/LeanCode.ContractsGenerator/${SERVER_VERSION}"
```

Comment:
was `/.local/state` intentional?

Reply:
Yup, that's the default for `XDG_STATE_HOME` as per [XDG Base Directory Spec](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html):

>  `$XDG_STATE_HOME` defines the base directory relative to which user-specific state files should be stored. If `$XDG_STATE_HOME` is either not set or empty, a default equal to `$HOME/.local/state` should be used.
> 
> The `$XDG_STATE_HOME` contains state data that should persist between (application) restarts, but that is not important or portable enough to the user that it should be stored in `$XDG_DATA_HOME`. It may contain:
> 
> * actions history (logs, history, recently used files, …)
> 
> * current state of the application that can be reused on a restart (view, layout, open files, undo history, …)

Reply:
TIL